bert/examples/onnx_runtime.rs

```diff
@@ -0,0 +1,169 @@
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use derive_more::Deref;
+use ndarray::Array2;
+use onnxruntime::tensor::OrtOwnedTensor;
+use onnxruntime::{environment::Environment, GraphOptimizationLevel, LoggingLevel};
+use serde::{Deserialize};
+use std::time::Duration;
+use tokenizers::{
+    tokenizer::Tokenizer as HfTokenizer, PaddingDirection, PaddingParams, PaddingStrategy,
+    TruncationDirection, TruncationParams, TruncationStrategy,
+};
+use xayn_ai_bert::{tokenizer::Tokenizer, Config, FirstPooler};
+use xayn_test_utils::asset::{smbert, sts_data, zdf_data};
+
+fn create_tokenizer(token_size: usize) -> Result<HfTokenizer, Box<dyn std::error::Error>> {
+    let config = Config::new(smbert()?)?.with_token_size(token_size)?;
+    let mut tokenizer = HfTokenizer::from_file(config.dir.join("tokenizer.json")).unwrap();
+    let padding_token = config.extract::<String>("tokenizer.tokens.padding")?;
+    let padding = PaddingParams {
+        strategy: PaddingStrategy::Fixed(token_size),
+        direction: PaddingDirection::Right,
+        pad_to_multiple_of: None,
+        pad_id: tokenizer
+            .token_to_id(&padding_token)
+            .ok_or("missing padding token")?,
+        pad_type_id: 0,
+        pad_token: padding_token,
+    };
+    let truncation = TruncationParams {
+        direction: TruncationDirection::Right,
+        max_length: token_size,
+        strategy: TruncationStrategy::LongestFirst,
+        stride: 0,
+    };
+    tokenizer.with_padding(Some(padding));
+    tokenizer.with_truncation(Some(truncation));
+
+    Ok(tokenizer)
+}
+
+fn run_onnx(texts: Texts, token_size: usize) -> Result<Duration, Box<dyn std::error::Error>> {
+    let tokenizer = create_tokenizer(token_size)?;
+    let token_reshaper =
+        |slice: &[u32]| Array2::from_shape_fn((1, slice.len()), |(_, i)| i64::from(slice[i]));
+
+    let environment = Environment::builder()
+        .with_name("test")
+        .with_log_level(LoggingLevel::Verbose)
+        .build()?;
+
+    let mut session = environment
+        .new_session_builder()?
+        .with_optimization_level(GraphOptimizationLevel::Extended)?
+        .with_number_threads(1)?
+        .with_model_from_file("/Users/maciejkrajewski/CLionProjects/xayn_discovery_engine/assets/smbert_v0004/model.onnx")?;
+
+    let start = std::time::Instant::now();
+    for text in texts.into_vec() {
+        let tokens = tokenizer.encode(text, false).unwrap();
+        // reshape input tensor
+        let token_ids = token_reshaper(tokens.get_ids());
+        let attention_mask = token_reshaper(tokens.get_attention_mask());
+        let type_ids = token_reshaper(tokens.get_type_ids());
+        let in_tensor = vec![token_ids, attention_mask, type_ids];
+        let outputs: Vec<OrtOwnedTensor<f32, _>> = session.run(in_tensor)?;
```

Comment:
depending internal details of  `OrtOwnedTensor<f32, _>` and `session.run(in_tensor)?` the compiler could theoretically be able to deduce that `output` is not used `session` (besides costing time) has no side effects and then end up eliminating the while loop :upside_down_face: 

practically this is impossible to happen as `session` will call C-FFI so the compiler doesn't know anymore that it doesn't have side effects etc.

still because of this you want to to pass unused output of benchmarks through  `std::hint::black_box` (similar const input values, but we don't have them here)